src/strands/strands_api.js

```diff
@@ -162,21 +161,27 @@ function createHookArguments(strandsContext, parameters){
   const dag = strandsContext.dag;
   
   for (const param of parameters) {
-    const paramType = param.type;
-    if(isStructType(paramType.typeName)) {
-      const structType = StructType[paramType.typeName];
-      const { id, dimension } = build.structInstanceNode(strandsContext, structType, param.name, []);
-      const structNode = createStrandsNode(id, dimension, strandsContext);//new StrandsNode(originalInstanceInfo.id, 0, strandsContext);
-
-      for (let i = 0; i < structType.properties.length; i++) {
-        const componentTypeInfo = structType.properties[i];
-        Object.defineProperty(structNode, componentTypeInfo.name, {
+    if(isStructType(param.type.typeName)) {
+      const structTypeInfo = structType(param);
+      const { id, dimension } = build.structInstanceNode(strandsContext, structTypeInfo, param.name, []);
+      const structNode = createStrandsNode(id, dimension, strandsContext);
+      for (let i = 0; i < structTypeInfo.properties.length; i++) {
+        const propertyType = structTypeInfo.properties[i];
+        Object.defineProperty(structNode, propertyType.name, {
           get() {
             const propNode = getNodeDataFromID(dag, dag.dependsOn[structNode.id][i])
-            return createStrandsNode(propNode.id, propNode.dimension, strandsContext);
+            const onRebind = (newFieldID) => {
```

Comment:
Worth reiterating what rebinding is for here too from the member assignment section